### PR TITLE
ann_catton introspection

### PR DESCRIPTION
**Pull request by anncatton**

### CODE REVIEW

File: _posts/ann_catton/2019-03-25-introspection.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            }
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+
+This was all the field data we needed for our new ui features. As a bonus, we now have a better idea of where some of our data is coming from - it was certainly surprising to find it was not all coming from our Elasticsearch models, as I had expected. And best of all, we have a strategy to integrate new clinical types that haven’t yet been indexed. By querying the schema itself, we’ll be able to pick them up as they are added.
```
Comment (b-f-chan) [inline _posts/ann_catton/2019-03-25-introspection.md]: Capitalize 'UI'

File: _posts/ann_catton/2019-03-25-introspection.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            name
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+However, the result didn’t go very far down the tree; what was under `"Case"` was mostly metadata fields, and we didn’t find a lot of the fields we were looking for from the models. We dug around some more, trying instead more specific types from our clinical data, such as `"Treatment"` and `"Diagnosis"`. I thought we'd hit the jackpot with this one; it returned a lot of data that _looked_ like what we needed - a good list of fields with their descriptions and types, without a lot of metadata fields we didn’t need. But comparing the results to our models, we soon discovered this was coming from our [data dictionary](https://github.com/NCI-GDC/gdcdictionary/tree/develop/gdcdictionary/schemas){:target="\_blank"}, and had more fields than what was in the index. We’d end up with a different version of the same issue. We needed to query a type that had its source in our case index.
```
Comment (b-f-chan) [inline _posts/ann_catton/2019-03-25-introspection.md]: "_...without a lot of metadata fields we didn't need._"

Sorry, I know @francisgerthoffert made you change this already, but the grammar confused me a bit here, I had to re-read it a couple times for it to click.  I think "...without a lot of..." was a bit of a mouthful.

Not a big deal and it's quite subjective, but maybe an alternative like, "_...but without metadata fields that we didn't need._"

P.S. I liked the original "ruder" version better :P.

File: _posts/ann_catton/2019-03-25-introspection.md (hunk 1 above)
Comment (anncatton) [reply _posts/ann_catton/2019-03-25-introspection.md]: right-o!

File: _posts/ann_catton/2019-03-25-introspection.md (hunk 2 above)
Comment (anncatton) [reply _posts/ann_catton/2019-03-25-introspection.md]: Hehe. I like your version, it's cleaner!

**Review by b-f-chan — APPROVED** [review]

**Pull request merged**